### PR TITLE
Adding TS types for pkg exports

### PR DESCRIPTION
**Pull request by rschristian**
This PR simply adds TS types for the package. Useful for TS and non-TS users alike. 

Let me know if any of these look incorrect.

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+type Styles = string | TemplateStringsArray | string[] | Record<string, boolean>;
+
+declare function process(theme: Record<string, unknown>): (strings: Styles) => Record<string, unknown>;
+declare function configure(theme: Record<string, unknown>): Record<string, unknown>;
+declare function themed(theme?: Record<string, unknown>): (strings: Styles) => string;
```
Comment (lukejacksonn) [inline index.d.ts:5]: I'm no TS expert but doesn't calling themed return a `process`?

Comment (rschristian) [reply index.d.ts:5]: Do you mean that themed should return a `process` or that it does?

Comment (rschristian) [reply index.d.ts:5]: Unless I'm misinterpreting, calling `themed` returns a function that takes `Styles` as a param and returns a `string` for the class name.

```
export const themed = (theme = {}) => {
  const processWithTheme = process(configure(theme));
  return (strings, ...values) => css(processWithTheme(strings, values));
};
```

Comment (lukejacksonn) [reply index.d.ts:5]: Ahh yeh, my bad, I misread the signature. In that case.. this looks good!

**Comment by lukejacksonn** [comment]
Thanks for this @rschristian and for you patience waiting for a merge 🙇

**Pull request merged**